pylint/checkers/classes.py

```diff
@@ -908,6 +908,13 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
             function_def = cast(astroid.FunctionDef, function_def)
             if not is_attr_private(function_def.name):
                 continue
+            if isinstance(function_def.parent.scope(), astroid.FunctionDef):
+                # Handle nested functions
+                outer_fn = function_def.parent.scope()
+                if function_def.name in [
+                    n.name for n in outer_fn.nodes_of_class(astroid.Name)
+                ]:
```

Comment:
```suggestion
                if function_def.name in (
                    n.name for n in outer_fn.nodes_of_class(astroid.Name)
                ):
```
You should be able to use a generator here

Reply:
pardon for my lack of experience here, but the suggestion is preferred because it's immutable right?

@cdce8p

Reply:
Not exactly. Here it's the difference between a list comprehension and a generator expression. The later can be slightly faster in some cases as it doesn't need to create a new list before iterating. Especially since `nodes_of_class` also returns a generator.

Immutability would be the case if it were normal list / tuple. Then a tuple would be preferred. (There is a pylint check for it in the CodeStyle extension).